library/modules/Job.cpp

```diff
@@ -608,15 +608,16 @@ bool Job::isSuitableItem(df::job_item *item, df::item_type itype, int isubtype)
     return iinfo.isValid() && iinfo.matches(*item, &minfo);
 }
 
-bool Job::isSuitableMaterial(df::job_item *item, int mat_type, int mat_index)
+bool Job::isSuitableMaterial(
+    df::job_item *item, int mat_type, int mat_index, df::item_type itype)
 {
     CHECK_NULL_POINTER(item);
 
     if (mat_type == -1 && mat_index == -1)
         return true;
 
     ItemTypeInfo iinfo(item);
-    MaterialInfo minfo(mat_type, mat_index);
+    MaterialInfo minfo(mat_type, mat_index, itype);
```

Comment:
Since `job_item.item_type` doesn't seem to be as useful as I had hoped, I think an alternative "clean" solution might be to pass `itype` to `matches()` below instead of storing it in `MaterialInfo`. (I think this is the only place where the stored `itype` is actually relied on, but I could be missing something - most of the other additions appear to just be passing it along.)